WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1550,10 +1551,17 @@ private void showGutenbergInformativeDialog() {
     }
 
     private void setGutenbergEnabledIfNeeded() {
+        boolean newSitePopupNeeded = AppPrefs.shouldShowGutenbergInfoPopup(mSite.getUrl());
         if ((TextUtils.isEmpty(mSite.getMobileEditor()) && !mIsNewPost)
-            || AppPrefs.shouldShowGutenbergInfoPopup(mSite.getUrl())) {
+            || newSitePopupNeeded) {
             SiteUtils.enableBlockEditor(mDispatcher, mSite);
-            AnalyticsUtils.trackWithSiteDetails(Stat.EDITOR_GUTENBERG_ENABLED, mSite);
+
+            if (!newSitePopupNeeded) {
+                // Don't track the activation of the block editor on new sites.
+                // The flip to GB happened at site creation time, and the flip is tracked there with proper details
```

Comment:
Hmm, should the `SiteUtils.enableBlockEditor(mDispatcher, mSite);` (line number 1557 above) be inside this conditional too then? 🤔

Reply:
Well, I think it's better to keep it out, since the user can change the option back to Aztec from settings right after the site creation. Even before creating/opening a new post. It's an edge case I know, but better to show both the dialog and repeat the network call to the server. It's just a call, that doesn't change anything on new sites 99% of the cases.

Reply:
As per chat convo - cleaned up the code an made more easy to follow in https://github.com/wordpress-mobile/WordPress-Android/pull/10300/commits/4db82286865a7cfc3b0c21c8e868ccd757db112e